File: scripts/import-literary-samples.ts
/**
 * 导入 export-literary-samples 生成的 manifest + public 资源到当前 DATABASE_URL。
 *
 *   npx tsx scripts/import-literary-samples.ts --in ./data/literary-samples-bundle
 */
import fs from "node:fs/promises";
import path from "node:path";
import { prisma } from "@/lib/prisma";

type Manifest = {
  version: number;
  ownerKey: string;
  novels: Array<Record<string, unknown>>;
  comics: Array<Record<string, unknown>>;
  files: string[];
};

async function main(): Promise<void> {
  const inArg = process.argv.indexOf("--in");
  const inDir = path.resolve(
    inArg >= 0 && process.argv[inArg + 1] ? process.argv[inArg + 1] : "data/literary-samples-bundle",
  );
  const repoRoot = process.cwd();
  const raw = await fs.readFile(path.join(inDir, "manifest.json"), "utf8");
  const manifest = JSON.parse(raw) as Manifest;

  let novelUpsert = 0;
  for (const row of manifest.novels) {
    const id = String(row.id);
    const data = {
      ownerKey: String(row.ownerKey ?? manifest.ownerKey),
      visibility: String(row.visibility ?? "public"),
      featured: Boolean(row.featured),
      shareCode: row.shareCode != null ? String(row.shareCode) : null,
      title: String(row.title),
      prompt: String(row.prompt),
      content: String(row.content),
      summary: row.summary != null ? String(row.summary) : null,
      lengthTier: row.lengthTier != null ? String(row.lengthTier) : "medium",
      generationMetaJson: row.generationMetaJson != null ? String(row.generationMetaJson) : null,
      creativeBriefJson: row.creativeBriefJson != null ? String(row.creativeBriefJson) : null,
      characterRosterJson: row.characterRosterJson != null ? String(row.characterRosterJson) : null,
      status: String(row.status ?? "ready"),
      coverPath: row.coverPath != null ? String(row.coverPath) : null,
      playCount: Number(row.playCount ?? 0),
      likeCount: Number(row.likeCount ?? 0),
      createdAt: row.createdAt ? new Date(String(row.createdAt)) : undefined,
      updatedAt: row.updatedAt ? new Date(String(row.updatedAt)) : undefined,
    };
    await prisma.novel.upsert({
      where: { id },
      create: { id, ...data },
      update: data,
    });
    novelUpsert++;
  }

  let comicUpsert = 0;
  for (const row of manifest.comics) {
    const id = String(row.id);
    const data = {
      ownerKey: String(row.ownerKey ?? manifest.ownerKey),
      visibility: String(row.visibility ?? "public"),
      featured: Boolean(row.featured),
      novelId: row.novelId != null ? String(row.novelId) : null,
      shareCode: row.shareCode != null ? String(row.shareCode) : null,
      title: String(row.title),
      prompt: String(row.prompt),
      imageUrls: String(row.imageUrls ?? "[]"),
      creativeBriefJson: row.creativeBriefJson != null ? String(row.creativeBriefJson) : null,
      status: String(row.status ?? "ready"),
      coverPath: row.coverPath != null ? String(row.coverPath) : null,
      likeCount: Number(row.likeCount ?? 0),
      createdAt: row.createdAt ? new Date(String(row.createdAt)) : undefined,
      updatedAt: row.updatedAt ? new Date(String(row.updatedAt)) : undefined,
    };
    await prisma.comic.upsert({
      where: { id },
      create: { id, ...data },
      update: data,
    });
    comicUpsert++;
  }

  let fileCopied = 0;
  for (const rel of manifest.files) {
    const src = path.join(inDir, "public", rel);
    const dest = path.join(repoRoot, "public", rel);
    try {
      await fs.mkdir(path.dirname(dest), { recursive: true });
      await fs.copyFile(src, dest);
      fileCopied++;
    } catch {
      console.warn("[warn] skip file:", rel);
    }
  }

  console.log(
    `[import-literary-samples] novels=${novelUpsert} comics=${comicUpsert} files=${fileCopied} owner=${manifest.ownerKey}`,
  );
}

main()
  .catch((e) => {
    console.error("[FAIL]", e instanceof Error ? e.message : e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: scripts/import-literary-samples.py
#!/usr/bin/env python3
"""Import manifest.json + public files into prod SQLite (upsert Novel/Comic)."""
import json, os, shutil, sqlite3, sys
from pathlib import Path

def main():
    in_dir = Path(sys.argv[sys.argv.index("--in") + 1]) if "--in" in sys.argv else Path("data/literary-samples-bundle")
    repo = Path.cwd()
    manifest = json.loads((in_dir / "manifest.json").read_text(encoding="utf-8"))
    db_path = os.environ.get("DATABASE_URL", "file:./prod.db").replace("file:", "").strip("./")
    db = repo / db_path
    db = str(db)

    c = sqlite3.connect(db)
    c.row_factory = sqlite3.Row

    def upsert(table, row, pk="id"):
        cols = list(row.keys())
        placeholders = ",".join("?" * len(cols))
        col_names = ",".join(cols)
        vals = []
        for k in cols:
            v = row[k]
            if isinstance(v, bool):
                v = 1 if v else 0
            vals.append(v)
        c.execute(
            f"INSERT OR REPLACE INTO {table} ({col_names}) VALUES ({placeholders})",
            vals,
        )

    for n in manifest["novels"]:
        upsert("Novel", n)
    for cm in manifest["comics"]:
        upsert("Comic", cm)
    c.commit()

    copied = 0
    for rel in manifest.get("files", []):
        src = in_dir / "public" / rel.replace("/", os.sep)
        dest = repo / "public" / rel.replace("/", os.sep)
        if not src.is_file():
            print("[warn] skip", rel)
            continue
        dest.parent.mkdir(parents=True, exist_ok=True)
        shutil.copy2(src, dest)
        copied += 1

    print(f"[ok] novels={len(manifest['novels'])} comics={len(manifest['comics'])} files={copied} db={db}")

if __name__ == "__main__":
    main()
